Assets/Puppet/PuppetAuth.cs

```diff
@@ -35,28 +35,23 @@ private IEnumerator OnEnableCoroutine()
         AuthEnabled.isOn = task.Result;
     }
 
-    public void SetEnabled(bool enabled)
-    {
-        StartCoroutine(SetEnabledCoroutine(enabled));
-    }
-
-    public void SignIn()
+    private IEnumerator SignInCoroutine()
     {
         var signInTask = Auth.SignInAsync();
-        signInTask.ContinueWith(t =>
-        {
-            if (t.Exception != null)
+            if (signInTask.Exception != null)
             {
+            yield return signInTask.Exception;
                 AuthStatus.text = UserNotAuthenticated;
                 AccountIdName.enabled = enabled;
                 AccountIdName.text = "Exception";
                 AccountIdLabel.enabled = enabled;
-                AccountIdLabel.text = t.Exception.ToString();
+                AccountIdLabel.text = signInTask.Exception.ToString();
             }
             else
```

Comment:
Do we have `t.IsCompleted` property defined? I guess we should check that each time and return a WaitForSeconds(timeout) so that we never actually block frame completion by waiting on a long running task

Reply:
Actually we have this logic in custom coroutines. Just I implemented it incorrect. Fixed.